contrib/endpoints/src/api_manager/proto/server_config.proto

```diff
@@ -196,6 +194,15 @@ message ServiceManagementConfig {
   int32 refresh_interval_ms = 2;
 }
 
+// Maps service configuration files to their corresponding traffic percentage.
+message ServiceConfigTrafficPercentage {
+  // The service configuration file name
+  string service_config = 1;
+  // The traffic percentage which must be greater than 0.0 and the sum must
+  // equal to 100.0.
```

Comment:
no, we don't need to add such restriction.  let is call it 

traffic_weight.   

the traffic is spitted according to the weights. sum of them not need to be 100%